### ISSUE TITLE
Waveform Data Creation

### ISSUE THREAD

**Issue by joncameron** [avalonmediasystem/avalon#3005]
### Description

Create the waveform data to be consumed by the frontend.  Target BBC audiowaveform style json.

Needs to be able to easily change to pull source master file from either derivative m3u8 http url or s3 path (or maybe master file on disk).

### Tasks
- [x] Evaluate existing solutions for generating waveform data and make a decision
- [x] Write service class which creates the waveform data
- [x] Add ActiveFedora::File to MasterFile model for waveform data file
- [x] Write job class which calls service class and stores into Fedora
- [x] Enqueue job with either http derivative url ~(or s3 path)~ (AWS work split out to its own story: #3018)
- [x] (Optional) Port waveform creation code to ruby
- [x] Handle job race condition
- [x] Write tests
~- [ ] Write documentation for any new configuration~

### Future work
- Backfill for existing master files (#3008)
~- Include into ingest api (#3007)~ - COMPLETED

Child of samvera-labs/avalon-bundle#177

**Comment by cjcolvar** [comment]
The backend is done for now but may need tweaks based upon the front end work.  I'm inclined to wait on the documentation since we don't have any configuration points right now, but we may after discovering the need for it in the front end work.
